From 39562bf062804f84dc84fea43ea8903dfd8daedf Mon Sep 17 00:00:00 2001
From: Michael Tretter <m.tretter@pengutronix.de>
Date: Wed, 23 Jan 2019 12:50:40 +0100
Subject: [PATCH 30/45] glupload: prevent segfault when updating caps

When the glupload element renegotiates the caps, set_caps will reset the
method_impl to NULL, but the method will be kept. transform_caps tries
to use the method_impl to transform the caps, because a method is set,
but will segfault.

(cherry picked from commit 9c2379e00aa4dc4ed8a8d84303f5a1bc2333f0ca)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/gl/gstglupload.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 8a63a2cbb2..5b88ec91b5 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -1873,6 +1873,7 @@ _gst_gl_upload_set_caps_unlocked (GstGLUpload * upload, GstCaps * in_caps,
   gst_video_info_from_caps (&upload->priv->in_info, in_caps);
   gst_video_info_from_caps (&upload->priv->out_info, out_caps);
 
+  upload->priv->method = NULL;
   upload->priv->method_impl = NULL;
   upload->priv->method_i = 0;
 
-- 
2.11.0

